zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/ElasticsearchStorage.java

```diff
@@ -101,14 +101,14 @@ public static Builder newBuilder() {
   @AutoValue.Builder
   public abstract static class Builder extends StorageComponent.Builder {
     /**
-     * Customizes the {@link HttpClientBuilder} used when connecting to ElasticSearch. Mostly for
-     * testing.
+     * Customizes the {@link HttpClientBuilder} used when connecting to ElasticSearch. This is used
+     * by the server and tests to enable detailed logging and tweak timeouts.
```

Comment:
this is good but doesnt explain why it needs to be public. usually for tests we use internal package accessors which allows us to remove or change the method.  we dont expose public methods unless there is non test code using it in other words

Reply:
ex below you could add exactly why like // specifically this is used for self tracing

or whatever it is used for.. reason is, that sometimes we have added methidd for gcp or aws and forgot and then broke them

Reply:
The main reason is `the server` - it calls these from `zipkin2.server.internal.elasticsearch`. Actually don't see usage of these in `zipkin-aws`.

I'll leave the comments like this for now since not sure of a better alternative.